File: backend/schemas/performer_song_selection.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime
from models.performer import PerformerStatus


class PerformerSongSelectionBase(BaseModel):
    selection_order: str
    is_singing: bool
    instrument: Optional[str] = None
    status: PerformerStatus


class PerformerSongSelectionCreate(PerformerSongSelectionBase):
    performer_id: int
    song_id: int


class PerformerSongSelectionUpdate(BaseModel):
    selection_order: Optional[str] = None
    is_singing: Optional[bool] = None
    instrument: Optional[str] = None
    status: Optional[PerformerStatus] = None


class PerformerSongSelectionResponse(PerformerSongSelectionBase):
    performer_selection_id: int
    performer_id: int
    song_id: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src_working/types/apiTypes/performerSongSelection.ts
// Performer Song Selection types matching backend schemas
import { PerformerStatus } from './performer';

export interface PerformerSongSelection {
    performer_selection_id: number;
    performer_id: number;
    song_id: number;
    selection_order: string;
    is_singing: boolean;
    instrument?: string;
    status: PerformerStatus;
    created_at: string;
}

export interface PerformerSongSelectionCreate {
    performer_id: number;
    song_id: number;
    selection_order: string;
    is_singing: boolean;
    instrument?: string;
    status: PerformerStatus;
}

export interface PerformerSongSelectionUpdate {
    selection_order?: string;
    is_singing?: boolean;
    instrument?: string;
    status?: PerformerStatus;
}
